src/renderers/shared/shared/ReactInstanceMap.js

```diff
@@ -26,19 +26,19 @@ var ReactInstanceMap = {
    * supported we can rename it.
    */
   remove: function(key) {
-    key._reactInternalInstance = undefined;
+    key._reactInternalFiber = undefined;
```

Comment:
This will break hot reloading (which we just fixed). Do you mind sending a PR to `react-deep-force-update` to account for this change? Should be as easy as checking both fields.